src/wui/interactive_base.cc

```diff
@@ -789,47 +737,21 @@ void InteractiveBase::draw_overlay(RenderTarget& dst) {
 
 	// Node information
 	std::string node_text;
-	if (game == nullptr) {
-		// Always blit node information in the editor
+	if (game == nullptr || get_display_flag(dfDebug)) {
+		// Blit node information in the editor, and in debug mode also for games
 		static boost::format node_format("(%i, %i, %i)");
 		const int32_t height = egbase().map()[sel_.pos.node].get_height();
 		node_text = (node_format % sel_.pos.node.x % sel_.pos.node.y % height).str();
-	} else if (get_display_flag(dfDebug)) {
-		// Blit node information for games in debug mode - we're not interested in the height
-		static boost::format node_format("(%i, %i)");
-		node_text = (node_format % sel_.pos.node.x % sel_.pos.node.y).str();
-	}
-	if (!node_text.empty()) {
-		std::shared_ptr<const UI::RenderedText> rendered_text = UI::g_fh->render(
-		   as_richtext_paragraph(node_text, UI::FontStyle::kWuiGameSpeedAndCoordinates));
-		rendered_text->draw(
-		   dst, Vector2i(get_w() - 5, get_h() - rendered_text->height() - 5), UI::Align::kRight);
 	}
+	info_panel_.set_coords_string(node_text);
 
 	// In-game clock and FPS
-	if ((game != nullptr) && get_config_bool("game_clock", true)) {
-		// Blit in-game clock
-		const std::string gametime(gametimestring(egbase().get_gametime().get(), true));
-		std::shared_ptr<const UI::RenderedText> rendered_text = UI::g_fh->render(
-		   as_richtext_paragraph(gametime, UI::FontStyle::kWuiGameSpeedAndCoordinates));
-		rendered_text->draw(dst, Vector2i(5, 5));
-
-		// Blit FPS when playing a game in debug mode
-		if (get_display_flag(dfDebug)) {
-			static boost::format fps_format("%5.1f fps (avg: %5.1f fps)");
-			rendered_text = UI::g_fh->render(
-			   as_richtext_paragraph((fps_format % (1000.0 / frametime_) % average_fps()).str(),
-			                         UI::FontStyle::kWuiGameSpeedAndCoordinates));
-			rendered_text->draw(dst, Vector2i((get_w() - rendered_text->width()) / 2, 5));
-		}
-	}
-
-	if (cheat_mode_enabled_) {
-		std::shared_ptr<const UI::RenderedText> rendered_text = UI::g_fh->render(
-		   as_richtext_paragraph("‹‹‹ CHEAT MODE ENABLED ›››", UI::FontStyle::kFsMenuIntro));
-		rendered_text->draw(
-		   dst, Vector2i((get_w() - rendered_text->width()) / 2, 2.5f * rendered_text->height()));
-	}
+	info_panel_.set_time_string(game ? gametimestring(egbase().get_gametime().get(), true) : "");
+	info_panel_.set_fps_string(
+	   get_display_flag(dfDebug) ?
+	      (boost::format("%5.1f fps (avg: %5.1f fps)") % (1000.0 / frametime_) % average_fps())
+	         .str() :
+	      "");
```

Comment:
Maybe show average fps only when the window has a minimum width. In 800x600 it overlaps with the coordinates.